tests/validation/tests/v3_api/test_multi_cluster_app.py

```diff
@@ -83,40 +37,375 @@ def test_multi_cluster_app_create():
+        templateVersionId=WORDPRESS_TEMPLATE_VID_2112,
+        targets=targets,
+        roles=ROLES,
+        name=random_name(),
+        answers=[{"values": answer_values}])
+    multiclusterapp = wait_for_mcapp_to_active(client, multiclusterapp)
+    validate_multi_cluster_app_cluster_wordpress(multiclusterapp)
+    delete_multi_cluster_app(multiclusterapp)
+
+
+def test_multi_cluster_app_template_rollback():
+    assert_if_valid_cluster_count()
     targets = []
     for projectid in project:
         targets.append({"projectId": projectid, "type": "target"})
-    temp_ver = "cattle-global-data:library-wordpress-1.0.5"
-    multiclusterapp = client.create_multiClusterApp(templateVersionId=temp_ver,
-                                                    targets=targets,
-                                                    roles=ROLES,
-                                                    name=random_name(),
-                                                    answers=[answer_105version]
-                                                    )
-    multiclusterapp = wait_for_mcapp_to_active(client, multiclusterapp)
-    # verify if this app is available in the cluster/project
-    validate_multi_cluster_app_cluster(multiclusterapp)
-    temp_ver = "cattle-global-data:library-wordpress-2.1.10"
-    multiclusterapp = client.update(multiclusterapp, uuid=multiclusterapp.uuid,
-                                    templateVersionId=temp_ver,
+    client = global_client["client"]
+    answer_values = get_defaut_question_answers(client, MYSQL_EXTERNALID_037)
+    multiclusterapp = client.create_multiClusterApp(
+        templateVersionId=MYSQL_TEMPLATE_VID_037,
+        targets=targets,
+        roles=ROLES,
+        name=random_name(),
+        answers=[{"values": answer_values}])
+    multiclusterapp = wait_for_mcapp_to_active(client, multiclusterapp)
+    validate_multi_cluster_app_cluster(multiclusterapp)
+    first_id = multiclusterapp["status"]["revisionId"]
+    assert multiclusterapp.templateVersionId == MYSQL_TEMPLATE_VID_037
+    answer_values_new = get_defaut_question_answers(
+        client, MYSQL_EXTERNALID_038)
+    multiclusterapp = client.update(multiclusterapp,
                                     roles=ROLES,
-                                    answers=[answer])
+                                    templateVersionId=MYSQL_TEMPLATE_VID_038,
+                                    answers=[{"values": answer_values_new}])
+    multiclusterapp = client.reload(multiclusterapp)
     multiclusterapp = wait_for_mcapp_to_active(client, multiclusterapp)
-    # verify if this app is available in the cluster/project
+    validate_multi_cluster_app_cluster(multiclusterapp)
+    assert multiclusterapp.templateVersionId == MYSQL_TEMPLATE_VID_038
+    # validate_app_upgrade_mca(multiclusterapp)
+    client.action(obj=multiclusterapp,
+                  action_name='rollback',
+                  revisionId=first_id)
```

Comment:
Can this be changed to multiclusterapp.rollback(first_id)?

Reply:
Done